waspc/e2e-test/test-outputs/waspComplexTest-golden/waspComplexTest/.wasp/out/sdk/wasp/dist/server/jobs/mySpecialJob.d.ts

```diff
@@ -16,7 +16,7 @@ export declare const mySpecialJob: {
         readonly pgBoss: {
             readonly cancel: () => ReturnType<import("pg-boss")["cancel"]>;
             readonly resume: () => ReturnType<import("pg-boss")["resume"]>;
-            readonly details: () => Promise<(Omit<import("pg-boss").JobWithMetadata<JSONObject>, "output" | "state"> & {
+            readonly details: () => Promise<Omit<import("pg-boss").JobWithMetadata<JSONObject>, "output" | "state"> & {
```

Comment:
I'm not sure why these e2e test changes happened 🤷

Reply:
Probably due to some Typescript emit fixes: https://github.com/microsoft/TypeScript/releases

Reply:
What do you mean by "emit fixes?"

Does this change our public API?

Reply:
Typescript keeps refining and fixing what they emit in `d.ts` files from version to version. Since we type our public API explicitly, I don't think it affects our public API since it mostly relates to the types Typescript infers (99% sure).

Reply:
Jobs are not yet explicitly typed (unfortunately, see https://github.com/wasp-lang/wasp/issues/1958), but I can't tell whether this affects the public API.

So, if you're relying on us explicitly typing the API, please double check.

Reply:
In this change we are relying on the `pg-boss` library types, that's not something we can type explicitly IMHO. So, it will change from time to time in the `d.ts` files. Or should we copy their types and use them directly instead of importing?

Reply:
So it does change the public API?

To avoid confusion, I don't know whether that's the case, I was just speculating.

Reply:
Ok, now that I look at the actual change: it went from `Promise<T | null>` to `Promise<T>` which means PgBoss types couldn't affected this. This means it's pure TS compiler related change. It looks like they were fixing a bug between 5.5.3 and 5.5.4.

```
npm install typescript@5.5.3
npx tsc --declaration --emitDeclarationOnly --stripInternal --declarationDir dist
```
diffed with
```
npm install typescript@5.5.4
npx tsc --declaration --emitDeclarationOnly --stripInternal --declarationDir dist
```
give me the diff we have here.

Reply:
I still don't know whether it changes our public API :smile: 

Does it?

Reply:
Yes, our public API changes between various TS versions due to their regressions and fixes. To avoid this, we should pin our TS version we use to build the SDK. But this is outside the scope of this PR.